### PR TITLE
Binary_Search in Kotlin

### LINKED ISSUE #1178: Binary Search in Kotlin
Please Assign it to me.@jainaman224

### PR DESCRIPTION
**Pull request by shubhammantri1**
Fixes #1178 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
- Adding Babylonian_Method in Kotlin

#### Languages Used:
- Kotlin

#### Files Added:
- Babylonian_Method.kt

### CODE REVIEW

File: Binary_Search/Binary_Search.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
+//Function For Binary Search
+fun binarySearch(list: List<Int>, left: Int, right: Int, num: Int):Int {
+    if (list.size >= 1) {
+        var mid = (left + (right -1)) / 2
```
Comment (MastersAbh) [inline Binary_Search/Binary_Search.kt]: A space after -

File: Binary_Search/Binary_Search.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        if(list[mid] == num) {
+            return mid
+        }
+        if(list[mid] > num) {
+            return binarySearch(list, 1, mid - 1, num)
+        }
+        if(list[mid] < num) {
+            return binarySearch(list, mid +1, right, num)
```
Comment (MastersAbh) [inline Binary_Search/Binary_Search.kt]: A space after +

File: Binary_Search/Binary_Search.kt [hunk 3]
```diff
@@ -0,0 +1,33 @@
+//Function For Binary Search
+fun binarySearch(list: List<Int>, left: Int, right: Int, num: Int):Int {
+    if (list.size >= 1) {
+        var mid = (left + (right -1)) / 2
+        if(list[mid] == num) {
```
Comment (MastersAbh) [inline Binary_Search/Binary_Search.kt]: space after if

File: Binary_Search/Binary_Search.kt [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+            return mid
+        }
+        if(list[mid] > num) {
```
Comment (MastersAbh) [inline Binary_Search/Binary_Search.kt]: space after if

File: Binary_Search/Binary_Search.kt [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+            return binarySearch(list, 1, mid - 1, num)
+        }
+        if(list[mid] < num) {
```
Comment (MastersAbh) [inline Binary_Search/Binary_Search.kt]: space after if

File: Binary_Search/Binary_Search.kt [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+            return binarySearch(list, mid +1, right, num)
+        }
+    }
+ return -1
+}
+
+//Driver Function for Binary Search
```
Comment (MastersAbh) [inline Binary_Search/Binary_Search.kt]: A space after //

File: Binary_Search/Binary_Search.kt [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+fun main() {
+    var list = mutableListOf(4, 6, 18, 29, 67, 89)
+    //let we want to find 67
```
Comment (MastersAbh) [inline Binary_Search/Binary_Search.kt]: A space after //

**Comment by somya-kapoor** [comment]
@shubhammantri1 Please take dynamic input.

File: Binary_Search/Binary_Search.kt (hunk 1 above)
Comment (shubhammantri1) [reply Binary_Search/Binary_Search.kt]: Done

File: Binary_Search/Binary_Search.kt (hunk 2 above)
Comment (shubhammantri1) [reply Binary_Search/Binary_Search.kt]: Done

File: Binary_Search/Binary_Search.kt (hunk 3 above)
Comment (shubhammantri1) [reply Binary_Search/Binary_Search.kt]: Done

File: Binary_Search/Binary_Search.kt (hunk 4 above)
Comment (shubhammantri1) [reply Binary_Search/Binary_Search.kt]: Done

File: Binary_Search/Binary_Search.kt (hunk 5 above)
Comment (shubhammantri1) [reply Binary_Search/Binary_Search.kt]: Done

File: Binary_Search/Binary_Search.kt (hunk 6 above)
Comment (shubhammantri1) [reply Binary_Search/Binary_Search.kt]: Done

File: Binary_Search/Binary_Search.kt (hunk 7 above)
Comment (shubhammantri1) [reply Binary_Search/Binary_Search.kt]: Done

**Comment by shubhammantri1** [comment]
Please Review!

**Comment by shubhammantri1** [comment]
Please Review @jainaman224

**Comment by jainaman224** [comment]
Squash your commits and it is ready to be merged.

**Comment by MastersAbh** [comment]
Hi @shubhammantri1 ,
If you need assistance with squashing of commits, you can reach out to us.

**Comment by rishabhgarg25699** [comment]
@shubhammantri1  You are on it or not . You only have to squash your commit .

**Comment by MastersAbh** [comment]
Hi @shubhammantri1 please squash you commits.
@rishabhgarg25699 @jainaman224 , you can take your decision on this PR. It's been a while now.

**Comment by MastersAbh** [comment]
@jainaman224 @rishabhgarg25699 There has been no activity on this PR. you can either merge it or close it. Only squashing is left.

**Comment by marwahmanbir** [comment]
I guess we can merge this one and close rest of the PRs by @shubhammantri1 due to inactivity.

**Pull request merged**